Metabase 0.49.8 update from 0.49.5 failed with clickhouse driver

### Describe the bug

Updating Metabase from 0.49.5 to 0.49.8 failed, with failed db migration.

### To Reproduce

1. make sure you run 0.49.5 version
2. try to upgrade metabase.jar with 0.49.8 version and restart it

### Expected behavior

Migration should be done withour problems

### Logs

 ```
 2024-05-07 10:07:44,852 INFO db.liquibase :: Running 1 migrations ...
 2024-05-07 10:07:45,220 INFO driver.impl :: Initializing driver :sql...
 2024-05-07 10:07:45,221 INFO driver.impl :: Initializing driver :sql-jdbc...
 2024-05-07 10:07:45,221 INFO driver.impl :: Initializing driver :postgres...
 2024-05-07 10:07:45,253 INFO driver.impl :: Initializing driver :h2...
 2024-05-07 10:07:45,259 INFO driver.impl :: Initializing driver :clickhouse...
 2024-05-07 10:07:45,260 INFO plugins.classloader :: Added URL file:/var/lib/metabase/plugins/clickhouse.metabase-driver.jar to classpath
 2024-05-07 10:07:45,275 DEBUG plugins.init-steps :: Loading plugin namespace metabase.driver.clickhouse...
 2024-05-07 10:07:45,300 ERROR liquibase.changelog :: ChangeSet migrations/001_update_migrations.yaml::v49.2024-04-09T10:00:03::qnkhuat encountered an exception.
 UPDATE SUMMARY
 Run:                          1
 Previously run:             268
 Filtered out:                17
 -------------------------------
 Total change sets:          286
 FILTERED CHANGE SETS SUMMARY
 DBMS mismatch:               17
 2024-05-07 10:07:45,355 ERROR metabase.core :: Metabase Initialization FAILED
 liquibase.exception.CommandExecutionException: java.lang.ClassCastException: class java.lang.ExceptionInInitializerError cannot be cast to class java.lang.Exception (java.lang.ExceptionInInitializerError and java.lang.Exception are in module java.base of loader 'bootstrap')
         at liquibase.command.CommandScope.execute(CommandScope.java:253)
         at liquibase.Liquibase.lambda$update$0(Liquibase.java:245)
         at liquibase.Scope.lambda$child$0(Scope.java:186)
         at liquibase.Scope.child(Scope.java:195)
         at liquibase.Scope.child(Scope.java:185)
         at liquibase.Scope.child(Scope.java:164)
         at liquibase.Liquibase.runInScope(Liquibase.java:1419)
         at liquibase.Liquibase.update(Liquibase.java:234)
         at liquibase.Liquibase.update(Liquibase.java:212)
         at liquibase.Liquibase.update(Liquibase.java:194)
         at metabase.db.liquibase$migrate_up_if_needed_BANG_.invokeStatic(liquibase.clj:305)
         at metabase.db.liquibase$migrate_up_if_needed_BANG_.invoke(liquibase.clj:287)
         at metabase.db.setup$migrate_BANG_$fn__51110.invoke(setup.clj:80)
         at metabase.db.liquibase$do_with_liquibase$f_STAR___48786.invoke(liquibase.clj:139)
         at metabase.db.liquibase$do_with_liquibase.invokeStatic(liquibase.clj:142)
         at metabase.db.liquibase$do_with_liquibase.invoke(liquibase.clj:130)
         at metabase.db.setup$migrate_BANG_.invokeStatic(setup.clj:75)
         at metabase.db.setup$migrate_BANG_.doInvoke(setup.clj:56)
         at clojure.lang.RestFn.invoke(RestFn.java:445)
         at metabase.db.setup$run_schema_migrations_BANG_.invokeStatic(setup.clj:147)
         at metabase.db.setup$run_schema_migrations_BANG_.invoke(setup.clj:141)
         at metabase.db.setup$setup_db_BANG_$fn__51138$fn__51139.invoke(setup.clj:165)
         at metabase.util.jvm$do_with_us_locale.invokeStatic(jvm.clj:239)
         at metabase.util.jvm$do_with_us_locale.invoke(jvm.clj:225)
         at metabase.db.setup$setup_db_BANG_$fn__51138.invoke(setup.clj:160)
         at metabase.db.setup$setup_db_BANG_.invokeStatic(setup.clj:159)
         at metabase.db.setup$setup_db_BANG_.invoke(setup.clj:153)
         at metabase.db$setup_db_BANG_$fn__51158.invoke(db.clj:69)
         at metabase.core$init_BANG__STAR_.invokeStatic(core.clj:116)
         at metabase.core$init_BANG__STAR_.invoke(core.clj:101)
         at metabase.core$init_BANG_.invokeStatic(core.clj:159)
         at metabase.core$init_BANG_.invoke(core.clj:154)
         at metabase.core$start_normally.invokeStatic(core.clj:171)
         at metabase.core$start_normally.invoke(core.clj:165)
         at metabase.core$entrypoint.invokeStatic(core.clj:204)
         at metabase.core$entrypoint.doInvoke(core.clj:198)
         at clojure.lang.RestFn.invoke(RestFn.java:397)
         at clojure.lang.AFn.applyToHelper(AFn.java:152)
         at clojure.lang.RestFn.applyTo(RestFn.java:132)
         at clojure.lang.Var.applyTo(Var.java:705)
         at clojure.core$apply.invokeStatic(core.clj:667)
         at clojure.core$apply.invoke(core.clj:662)
         at metabase.bootstrap$_main.invokeStatic(bootstrap.clj:31)
         at metabase.bootstrap$_main.doInvoke(bootstrap.clj:28)
         at clojure.lang.RestFn.invoke(RestFn.java:397)
         at clojure.lang.AFn.applyToHelper(AFn.java:152)
         at clojure.lang.RestFn.applyTo(RestFn.java:132)
         at metabase.bootstrap.main(Unknown Source)
 Caused by: java.lang.ClassCastException: class java.lang.ExceptionInInitializerError cannot be cast to class java.lang.Exception (java.lang.ExceptionInInitializerError and java.lang.Exception are in module java.base of loader 'bootstrap')
         at liquibase.command.CommandScope.logPrimaryExceptionToMdc(CommandScope.java:287)
         at liquibase.command.CommandScope.execute(CommandScope.java:245)
         ... 49 more
 2024-05-07 10:07:45,364 INFO metabase.core :: Metabase Shutting Down ...
 2024-05-07 10:07:45,371 INFO metabase.server :: Shutting Down Embedded Jetty Webserver
 2024-05-07 10:07:45,382 WARN db.liquibase :: ()
 2024-05-07 10:07:45,383 INFO metabase.core :: Metabase Shutdown COMPLETE
```

### Information about your Metabase installation

```JSON
{"file.encoding" "UTF-8",
  "java.runtime.name" "OpenJDK Runtime Environment",
  "java.runtime.version" "11.0.23+9",
  "java.vendor" "Eclipse Adoptium",
  "java.vendor.url" "https://adoptium.net/",
  "java.version" "11.0.23",
  "java.vm.name" "OpenJDK 64-Bit Server VM",
  "java.vm.version" "11.0.23+9",
  "os.name" "Linux",
  "os.version" "4.4.0-210-generic",
  "user.language" "en",
  "user.timezone" "UTC"}
```


### Severity

Blocking upgrade

### Additional context

_No response_

--- a/src/metabase/db/custom_migrations.clj
+++ b/src/metabase/db/custom_migrations.clj
@@ -1080,9 +1080,9 @@
   ;; this migraiton to remove triggers for any existing DB that have this option on.
   ;; See #40715
   (when-let [;; find all dbs which are configured not to scan field values
-             dbs (seq (filter #(and (-> % :details :let-user-control-scheduling)
+             dbs (seq (filter #(and (-> % :details json/parse-string :let-user-control-scheduling)
                                     (false? (:is_full_sync %)))
-                              (t2/select :model/Database)))]
+                              (t2/select :metabase_database)))]
     (classloader/the-classloader)
     (set-jdbc-backend-properties!)
     (let [scheduler (qs/initialize)]

--- a/src/metabase/db/custom_migrations.clj
+++ b/src/metabase/db/custom_migrations.clj
@@ -1080,7 +1080,7 @@
   ;; this migraiton to remove triggers for any existing DB that have this option on.
   ;; See #40715
   (when-let [;; find all dbs which are configured not to scan field values
-             dbs (seq (filter #(and (-> % :details json/parse-string :let-user-control-scheduling)
+             dbs (seq (filter #(and (-> % :details (json/parse-string keyword) :let-user-control-scheduling)
                                     (false? (:is_full_sync %)))
                               (t2/select :metabase_database)))]
     (classloader/the-classloader)

